### PR TITLE
Fix ImmutableArray.Builder.Sort(Comparer) to pass correct bounds

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/11173
cc: @CyrusNajmabadi, @AArnott

### CODE REVIEW

File: src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs [hunk 1]
```diff
@@ -288,10 +288,18 @@ public void Sort()
         [Fact]
         public void Sort_Comparison()
         {
-            var builder = new ImmutableArray<int>.Builder();
+            var builder = new ImmutableArray<int>.Builder(4);
```
Comment (CyrusNajmabadi) [inline src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs]: can you make these arrays of strings.  that way your comparer will null ref if you get passed a default element?

Comment (CyrusNajmabadi) [inline src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs]: (that's what happend in our case btw).

**Comment by CyrusNajmabadi** [comment]
Thanks for the speed on this.  :+1:

File: src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs]: I could.  But the tests I added catch the case, regardless (the values get sorted with 0s at the beginning, causing the comparison to fail), so I'm not sure doing so would add much.

**Comment by stephentoub** [comment]
> Thanks for the speed on this

:smile:

File: src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs]: ah, that's fair.  :)

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please (now that fix was merged)

**Pull request merged**